pkg/R/utils.R

```diff
@@ -259,3 +267,40 @@ joinTaggedList <- function(tagged_list, cnull) {
   lists <- genCompactLists(tagged_list, cnull)
   mergeCompactLists(lists[[1]], lists[[2]])
 }
+
+# Recursively examine variables in functions to decide if their values should
+# be included in the new function environment.
+closure.process <- function(
+  node,
+  func,
+  env
+) {
+  nlen <- length(node)
+  if(nlen > 1) {
+    for(i in 1:nlen) {
+      closure.process(node[[i]], func, env)
+    }
+  } else if(nlen == 1) {
+    if(mode(node) == 'name') {
+      cnode <- as.character(node)
+      if(!cnode %in% names(as.list(args(func)))) { # Not a function argument
+        func.env <- environment(func)  
+        if(exists(cnode, envir=func.env, inherits=F)) {
```

Comment:
I was trying not to take the variables in the package environment (like base). Please take a look at my new commit.